packages/one-app-runner/src/startApp.js

```diff
@@ -124,24 +157,92 @@ module.exports = async function startApp({
-  const devProxyServerPort = process.env.HTTP_ONE_APP_DEV_PROXY_SERVER_PORT || 3002;
-  const metricsPort = process.env.HTTP_METRICS_PORT || 3005;
-  const debugPort = process.env.HTTP_ONE_APP_DEBUG_PORT || 9229;
-  const ports = `-p ${appPort}:${appPort} -p ${devCDNPort}:${devCDNPort} -p ${devProxyServerPort}:${devProxyServerPort} -p ${metricsPort}:${metricsPort} -p ${debugPort}:${debugPort}`;
-
-  const command = `${generatePullCommand()} docker run -t ${ports} -e NODE_ENV=development ${generateContainerNameFlag()} ${generateNetworkToJoin()} ${generateEnvironmentVariableArgs(envVars)} ${generateModuleMountsArgs(modulesToServe)} ${generateCaCertsCommands(envVars)} ${appDockerImage} /bin/sh -c "${generateServeModuleCommands(modulesToServe)} ${generateSetMiddlewareCommand(parrotMiddlewareFile)} ${generateSetDevEndpointsCommand(devEndpointsFile)} node ${generateDebug(debugPort)} lib/server/index.js --root-module-name=${rootModuleName} ${generateModuleMap()} ${generateUseMocksFlag(parrotMiddlewareFile)} ${generateUseHostFlag()}"`;
-  const dockerProcess = spawn(command, { shell: true });
-  dockerProcess.on('error', () => {
+
+  const appPort = Number.parseInt(process.env.HTTP_PORT, 10) || 3000;
+  const devCDNPort = Number.parseInt(process.env.HTTP_ONE_APP_DEV_CDN_PORT, 10) || 3001;
+  const devProxyServerPort = Number.parseInt(
+    process.env.HTTP_ONE_APP_DEV_PROXY_SERVER_PORT, 10
+  ) || 3002;
+  const metricsPort = Number.parseInt(process.env.HTTP_METRICS_PORT, 10) || 3005;
+  const debugPort = Number.parseInt(process.env.HTTP_ONE_APP_DEBUG_PORT, 10) || 9229;
+
+  const ports = [
+    appPort,
+    devCDNPort,
+    devProxyServerPort,
+    metricsPort,
+    debugPort,
+  ];
+
+  const containerEnvVars = generateEnvironmentVariableArgs(envVars);
+
+  const mounts = new Map([
+    ...(modulesToServe || []).map((moduleToServe) => [moduleToServe, `/opt/module-workspace/${path.basename(moduleToServe)}`]),
+  ]);
+
+  const hostNodeExtraCaCerts = envVars.NODE_EXTRA_CA_CERTS || process.env.NODE_EXTRA_CA_CERTS;
+  if (hostNodeExtraCaCerts) {
+    console.log('mounting host NODE_EXTRA_CA_CERTS');
+    const mountPath = '/opt/certs.pem';
+    mounts.set(hostNodeExtraCaCerts, mountPath);
+    containerEnvVars.set('NODE_EXTRA_CA_CERTS', mountPath);
+  }
+
+  const containerShellCommand = `${
+    generateServeModuleCommands(modulesToServe)
+  } ${
+    generateSetMiddlewareCommand(parrotMiddlewareFile)
+  } ${
+    generateSetDevEndpointsCommand(devEndpointsFile)
+  } node ${
+    generateDebug(debugPort, useDebug)
+  } lib/server/index.js --root-module-name=${rootModuleName} ${
+    generateModuleMap(moduleMapUrl)
+  } ${
+    generateUseMocksFlag(parrotMiddlewareFile)
+  } ${
+    generateUseHostFlag()
+  }`;
+
+  const logFileStream = outputFile ? fs.createWriteStream(outputFile) : null;
+
+  const hostOneAppDirectoryPath = path.resolve(os.homedir(), '.one-app');
+  mounts.set(hostOneAppDirectoryPath, '/home/node/.one-app');
```

Comment:
could it be possible that these don't match what dev cdn is expecting ? should one-app accept a configuration argument to this? that way we can always ensure that they will match.

Reply:
resolving and this can be added in future update